model-api/src/commonMain/kotlin/org/modelix/model/api/IConcept.kt

```diff
@@ -181,3 +181,22 @@ interface IConcept {
 fun IConcept?.isSubConceptOf(superConcept: IConcept?) = this?.isSubConceptOf(superConcept) == true
 
 fun IConcept.conceptAlias() = getConceptProperty("alias")
+
+/**
+ * Checks if this is a sub-concept of [superConcept], based on the Concepts' UID.
+ *
+ * @param superConcept the potential super-concept
+ * @return true if this concept (or any of its ancestors) has the same UID as the [superConcept]
+ */
+fun IConcept.isSubConceptOfBasedOnUid(superConcept: IConcept): Boolean {
```

Comment:
```suggestion
fun IConcept.isSubConceptOfBasedOnUid(superConceptReference: IConceptReference): Boolean {
```
And then compare only by the UID.

Reply:
fixed in [f70040c](https://github.com/modelix/modelix.core/pull/819/commits/f70040c97d79bd6a0a9e77383c5bdc7844b3918f)